RealmSwift/Tests/RealmTests.swift

```diff
@@ -1953,66 +1954,239 @@ class LoggerTests: TestCase, @unchecked Sendable {
+            logs.withLock({ $0 += "\(level.logLevel) \(category.rawValue) \(message) " })
+        })
+        Logger.shared = logger
+
+        Logger.setLogLevel(.trace, for: Category.realm)
+        XCTAssertEqual(Logger.logLevel(for: Category.realm), .trace)
+
+        for category in Logger.allCategories() {
+            let categoryEnum = categoryfromString(category)
+            XCTAssertEqual(Logger.logLevel(for: categoryEnum!), .trace)
+        }
+
+        Logger.setLogLevel(.all, for: Category.realm)
+        XCTAssertEqual(Logger.logLevel(for: Category.realm), .all)
+
+        for category in Logger.allCategories() {
+            let categoryEnum = categoryfromString(category)
+            XCTAssertEqual(Logger.logLevel(for: categoryEnum!), .all)
+        }
+    }
+
+    func testChangeSubCategoryLevel() throws {
+        Logger.setLogLevel(.off, for: Category.realm)
+        XCTAssertEqual(Logger.logLevel(for: Category.Storage.all), .off)
+        XCTAssertEqual(Logger.logLevel(for: Category.Storage.transaction), .off)
+        XCTAssertEqual(Logger.logLevel(for: Category.Storage.query), .off)
+        XCTAssertEqual(Logger.logLevel(for: Category.Storage.object), .off)
+        XCTAssertEqual(Logger.logLevel(for: Category.Storage.notification), .off)
+
+        Logger.setLogLevel(.info, for: Category.Storage.all)
+        XCTAssertEqual(Logger.logLevel(for: Category.Storage.all), .info)
+        XCTAssertEqual(Logger.logLevel(for: Category.Storage.transaction), .info)
+        XCTAssertEqual(Logger.logLevel(for: Category.Storage.query), .info)
+        XCTAssertEqual(Logger.logLevel(for: Category.Storage.object), .info)
+        XCTAssertEqual(Logger.logLevel(for: Category.Storage.notification), .info)
+
+        XCTAssertEqual(Logger.logLevel(for: Category.realm), .off)
+        XCTAssertEqual(Logger.logLevel(for: Category.sdk), .off)
+        XCTAssertEqual(Logger.logLevel(for: Category.app), .off)
+        XCTAssertEqual(Logger.logLevel(for: Category.Sync.all), .off)
+    }
+
+    func testCallbackFilteringForCatgories() throws {
+        let logs = Locked("")
+        let logger = Logger(function: { level, _, message in
+            logs.withLock({ $0 += "\(Date.now) \(level.logLevel) \(message)" })
+        })
+
+        Logger.shared = logger
+
+        Logger.setLogLevel(.off, for: Category.realm)
+        Logger.setLogLevel(.info, for: Category.Storage.all)
+
+        logger.log(with: .info, categoryName: Category.Storage.all.rawValue, message: "Storage test entry")
+        XCTAssertTrue(logs.value.contains("Storage test entry"))
+        logs.wrappedValue = ""
+
+        logger.log(with: .info, categoryName: Category.Storage.transaction.rawValue, message: "Transaction test entry")
+        XCTAssertTrue(logs.value.contains("Transaction test entry"))
+        logs.wrappedValue = ""
+
+        logger.log(with: .info, categoryName: Category.realm.rawValue, message: "REALM test entry")
```

Comment:
Why is this passing the raw category name rather than the `LogCategory`?

Reply:
Because we cannot expose internal function to the test suite, instead we are using Realm.Private API to generate the logs for the tests